transmart-rest-api/src/main/groovy/org/transmartproject/rest/user/AuthContext.groovy

```diff
@@ -19,31 +18,11 @@ import org.transmartproject.core.users.UsersResource
 class AuthContext {
 
     @Autowired
-    private UsersResource usersResourceService
+    private UsersResource usersResource
 
     @Lazy
     User user = { ->
-        def userPrincipal = SecurityContextHolder.context.authentication?.principal
-        if (!userPrincipal) {
-            throw new AccessDeniedException('No user principal has found.')
-        }
-
-        if (userPrincipal instanceof User) {
-            log.debug('Principal is of the User type already. Returns the user.')
-            return userPrincipal
-        }
-
-        String username
-        if (userPrincipal instanceof String) {
-            log.debug("Principal is a '${userPrincipal}' string. Interpreting it as username.")
-            username = userPrincipal
-        } else if (userPrincipal.hasProperty('username')) {
-            log.debug("Principal has ${userPrincipal.class} type and has a username field.")
-            username = userPrincipal.username
-        } else {
-            throw new AccessDeniedException("Unsuported principal type: ${userPrincipal.class}.")
-        }
-        usersResourceService.getUserFromUsername(username)
+        usersResource.getUserFromPrincipal(SecurityContextHolder.context.authentication)
```

Comment:
this is single line that makes rest-api plugin depend on spring security library.
Considering to use `request.getuserprincipal()` instead and remove the dependency.

Reply:
And what if you don't have a request? E.g., asynchronous jobs or tests?